src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/VsContainedLanguageComponentsFactory.cs

```diff
@@ -68,30 +67,37 @@ private async Task<(HierarchyId itemid, IVsHierarchy? hierarchy, IVsContainedLan
             return (itemid, _projectVsServices.VsHierarchy, containedLanguageFactory);
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0030:Do not used banned APIs", Justification = "We are using IVsService and SwitchToUIThread")]
         private async Task<IVsContainedLanguageFactory?> GetContainedLanguageFactoryAsync()
         {
-            Guid languageServiceId = await GetLanguageServiceId();
-            if (languageServiceId == Guid.Empty)
+            Type languageServiceType = await GetLanguageServiceType();
+            if (languageServiceType == null)
                 return null;
 
             IOleAsyncServiceProvider serviceProvider = await _serviceProvider.GetValueAsync();
 
-            //TODO: Merged PIAs
-            object service = null;// = await serviceProvider.QueryServiceAsync(ref languageServiceId);
+            object? service = await serviceProvider.GetServiceAsync(languageServiceType);
 
             // NOTE: While this type is implemented in Roslyn, we force the cast on 
             // the UI thread because they are free to change this to an STA object
             // which would result in an RPC call from a background thread.
             await _projectVsServices.ThreadingService.SwitchToUIThread();
 
-            return (IVsContainedLanguageFactory)service;
+            if(null != service)
+            {
+                return (IVsContainedLanguageFactory)service;
+            }
+            else
+            {
+                return null;
+            }
```

Comment:
Couple things here:
- The cast (boxing) does nothing for a return statement in general. Here, it is there only because the return type is `IVsContainedLanguageFactory?` and the compiler needs to know how to convert to that. The non-nullable version, `IVsContainedLanguageFactory`, is convertable to the nullable version.
- This entire block can just be `return service as IVsContainedLanguageFactory;`. The if-else only exists because of my first bullet point. `as IVsContainedLanguageFactory` automatically is understood as `IVsContainedLanguageFactory?`.

```suggestion
            return service as IVsContainedLanguageFactory;
```